### PR TITLE
cpu:aarch64: Extending support for BRGEMM General and 1x1 Forward Convolution

### PR DESCRIPTION
**Pull request by kasturedeeksha**
# Description

This commit expands ARM SVE support for JIT SVE BRGEMM general and 1x1 forward convolution in FP32, introducing compatibility with various vector lengths. The changes made are for implementing different ARM SVE vector lengths.

Major code changes:
1. generalized code by adding template with isa as argument.
2. set predicate registers according to SVE length.
# Checklist

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit? Yes
Test output is same with and without this commit.
1. make test output: 
```
98% tests passed, 3 tests failed out of 195

Total Test time (real) = 1375.38 sec

The following tests FAILED:
        149 - test_graph_unit_dnnl_conv_usm_cpu (Failed)
        154 - test_graph_unit_dnnl_large_partition_usm_cpu (Failed)
        176 - test_benchdnn_modeC_graph_ci_cpu (Failed)
Errors while running CTest
```
- [x] Have you formatted the code using clang-format? Yes
cc : @kawakami-k

### CODE REVIEW

File: src/cpu/aarch64/brgemm/jit_brgemm_kernel.cpp [hunk 1]
```diff
@@ -1876,7 +1876,16 @@ void jit_brgemm_kernel_t::bdb_loop() {
 }
 
 void jit_brgemm_kernel_t::generate() {
+    size_t simd_w_ = (brg.isa_impl == sve_512)
```
Comment (dzarukin) [inline src/cpu/aarch64/brgemm/jit_brgemm_kernel.cpp:1879]: Nit: sticking to `switch` statement is safer and will allow to catch an error when another length will be added.

File: src/cpu/aarch64/jit_brgemm_conv.cpp [hunk 2]
```diff
@@ -634,7 +634,7 @@ status_t brgemm_convolution_fwd_t<isa, use_inversion>::add_po_kernel(
     bcfg->alpha = !is_init && IMPLICATION(jcp.with_sum, jcp.use_buffer);
     bcfg->beta = is_init ? 0 : 1;
     CHECK(safe_ptr_assign(kernels_po_[ker_idx],
-            new jit_brgemm_kernel_post_ops(jcp, *bcfg, *_pd->attr())));
+            new jit_brgemm_kernel_post_ops<isa>(jcp, *bcfg, *_pd->attr())));
```
Comment (dzarukin) [inline src/cpu/aarch64/jit_brgemm_conv.cpp:637]: I very highly encourage not to step on template path and pass length as argument to the function that crates a kernel or to a proxy that will dispatch templated kernel.

File: src/cpu/aarch64/jit_brgemm_post_ops.hpp [hunk 3]
```diff
@@ -196,7 +196,15 @@ struct jit_brgemm_kernel_diff_bias_t : public jit_generator {
     }
 
     void generate() override {
+        size_t simd_w_ = (brg_.isa_impl == sve_512)
```
Comment (dzarukin) [inline src/cpu/aarch64/jit_brgemm_post_ops.hpp:199]: Ditto.

File: src/cpu/aarch64/jit_brgemm_conv.cpp (hunk 2 above)
Comment (kasturedeeksha) [reply src/cpu/aarch64/jit_brgemm_conv.cpp:637]: Thank you for the feedback. Currently, we have implemented this using the template path. We will look into updating it to pass the length as an argument in future revisions.

File: src/cpu/aarch64/brgemm/jit_brgemm_kernel.cpp (hunk 1 above)
Comment (kasturedeeksha) [reply src/cpu/aarch64/brgemm/jit_brgemm_kernel.cpp]: Thank you for the suggestion. The switch statement has been implemented as recommended.

**Review by dzarukin — APPROVED** [review]

**Pull request merged**